packages/integration-tests/src/test-project-setup/data_storage_auth_with_triggers.ts

```diff
@@ -232,4 +253,27 @@ class DataStorageAuthWithTriggerTestProject extends TestProjectBase {
     // check expected response
     assert.deepStrictEqual(responsePayload, expectedResponse);
   };
+
+  private assertExpectedCleanup = async () => {
+    assert.equal(
+      await this.checkBucketExists(this.testBucketName),
+      false,
+      `Expected bucket ${this.testBucketName} to be removed but it still exists`
+    );
+  };
+
+  private checkBucketExists = async (bucketName: string): Promise<boolean> => {
+    try {
+      // there appears to be some eventual consistency between deleting a bucket and when HeadBucket returns NotFound
+      await new Promise((resolve) => setTimeout(resolve, 1000 * 30)); // wait 30 seconds
```

Comment:
if we increase wait we can also consider retries. this may help speed this up on average if eventual consistency happens earlier.

Reply:
Yeah, right now I'm just trying to figure out what a reasonable stabilization time looks like. I'm also considering removing this test altogether as it's really only testing CDK and if it's a super expensive test it's probably not worth it. Also depends on how strongly @Amplifiyer feels

Reply:
Looks like 30 seconds is enough time which isn't terrible. I'll add polling to speed it up in sunny day cases

Reply:
Not super strong, just a rule of thumb that any bug found should have automated testing.